sdk/textanalytics/Azure.AI.TextAnalytics/src/SentenceSentiment.cs

```diff
@@ -45,5 +50,55 @@ internal SentenceSentiment(SentenceSentimentInternal sentenceSentiment)
         /// for each sentiment. Higher values signify higher confidence.
         /// </summary>
         public SentimentConfidenceScores ConfidenceScores { get; }
+
+        /// <summary>
+        /// Gets the mined opinions of a sentence. This is only returned if
+        /// <see cref="AnalyzeSentimentOptions.IncludeOpinionMining"/> is set to true.
+        /// </summary>
+        public IReadOnlyCollection<MinedOpinion> MinedOpinions { get; }
+
+        private static IReadOnlyCollection<MinedOpinion> ConvertToMinedOpinions(SentenceSentimentInternal sentence, IReadOnlyList<SentenceSentimentInternal> allSentences)
+        {
+            var minedOpinions = new List<MinedOpinion>();
+
+            foreach (SentenceAspect aspects in sentence.Aspects)
+            {
+                var opinions = new List<OpinionSentiment>();
+                foreach (AspectRelation relation in aspects.Relations)
+                {
+                    if (relation.RelationType == AspectRelationType.Opinion)
+                    {
+                        opinions.Add(ResolveOpinionReference(allSentences, relation.Ref));
+                    }
+                }
+                minedOpinions.Add(new MinedOpinion(new AspectSentiment(aspects), opinions));
+            }
+
+            return minedOpinions;
+        }
+
+        private static Regex _opinionRegex = new Regex(@"/documents/(?<documentIndex>\d*)/sentences/(?<sentenceIndex>\d*)/opinions/(?<opinionIndex>\d*)$", RegexOptions.Compiled, TimeSpan.FromSeconds(2));
+        internal static OpinionSentiment ResolveOpinionReference(IReadOnlyList<SentenceSentimentInternal> sentences, string reference)
+        {
+            // Example: the following should result in sentenceIndex = 2, opinionIndex = 1. There will not be cases where sentences from other documents are referenced.
+            // "#/documents/0/sentences/2/opinions/1"
```

Comment:
Should we include the `#` in the regex then if other documents won't ever be referenced?  I'm not super worried about the perf though so I think it's fine to leave as-is if you think it's safe.

Reply:
Not sure I understand. Could you expand more on the ask?

Reply:
My understanding is this `#/documents/0/sentences/2/opinions/1` [JSON Pointer syntax](https://tools.ietf.org/html/rfc6901#section-6) is being used as a URI fragment in which case the `#` represents the current document.  The comment says it'll only ever refer to the same document, so I was wondering whether we should include matching the `#` character in the regex (and then anchoring at least the start of the regex with `^`) so we don't accidentally match other things if this is extended in the future.

While the official spec doesn't show it, most implementations allow external JSON Pointer links (including AutoRest with swagger files) where it would look like `myfile.json#/documents/0/sentences/2/opinions/1`.

The ask is whether or not we should be more defensive here around ensuring the JSON Pointer is aimed at only the current document.  I don't feel strongly if we're confident the service is always going to be returning this format.

Reply:
Nice! thanks for explaining it.
Because of the nature of the service and this endpoint, I will say we are ok as is. Results are always isolated to each document